test/integration_test.py

```diff
@@ -408,6 +409,22 @@ def __check_manifest(self, file: IO[bytes], uuid_field_name: str) -> List[Mappin
         self.assertEqual(bundle_uuid, str(uuid.UUID(bundle_uuid)))
         return rows
 
+    def _check_curl_manifest(self, _catalog: CatalogName, response: bytes):
+        text = TextIOWrapper(BytesIO(response))
+        urls, names = [], []
+        for line in text.readlines():
+            if line.startswith('--') or line == '':
+                pass  # skip empty lines and config lines like '--compressed\n'
+            elif m := re.fullmatch(r'url="(.+)"\n', line):
+                urls.append(m.group(1))
+                self.assertEqual(len(urls), len(names) + 1, line)
```

Comment:
I'm assuming the result looks something like:
```
---compressed
---foo

url="example.com"
output="the name"
url="foo.com"
output="the other name"
```
It might be more elegant to use `more_itertools.grouper()` to assert each line of the repeating chunk. Of course you would have factor the non-repeating parts out of the loop.

Let me know if this isn't making sense. Anyway this comment is really just a suggestion.